### PR TITLE
Remove SignaturesList type

### LINKED ISSUE #174: Remove SignaturesList
ℹ️ This issue is part of Epic #169 

## Objective

Apply the required changes for removing the **SignaturesList**.

### PR DESCRIPTION
**Pull request by EdwinGuayacan**
Fixes #174 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/types/signature.ex [hunk 1]
```diff
@@ -13,5 +13,6 @@ defmodule Kadena.Types.Signature do
 
   @impl true
   def new(sig) when is_binary(sig), do: %__MODULE__{sig: sig}
+  def new(sig) when is_nil(sig), do: []
```
Comment (miguelnietoa) [inline lib/types/signature.ex]: @EdwinGuayacan why is this required? why nil is not an invalid value?

File: lib/pact/command/cont_command.ex [hunk 2]
```diff
@@ -305,8 +305,8 @@ defmodule Kadena.Pact.ContCommand do
   end
 
   @spec build_signatures(sign_commands :: sign_commands(), result :: list()) :: valid_signatures()
-  defp build_signatures([], result), do: {:ok, SignaturesList.new(result)}
+  defp build_signatures([], result), do: {:ok, List.flatten(result)}
```
Comment (miguelnietoa) [inline lib/pact/command/cont_command.ex]: @EdwinGuayacan why is this required?

**Review by miguelnietoa — CHANGES_REQUESTED** [review]

File: lib/chainweb/pact/local_request_body.ex [hunk 3]
```diff
@@ -37,8 +37,8 @@ defmodule Kadena.Chainweb.Pact.LocalRequestBody do
   end
 
   @spec to_signature_list(signatures :: sigs()) :: {:ok, raw_sigs()}
-  defp to_signature_list(%SignaturesList{signatures: list}) do
-    sigs = Enum.map(list, fn sig -> Map.from_struct(sig) end)
+  defp to_signature_list(signatures) do
+    sigs = Enum.map(signatures, fn sig -> Map.from_struct(sig) end)
```
Comment (juanhurtado10) [inline lib/chainweb/pact/local_request_body.ex]: @EdwinGuayacan My 5 cents

```elixir
sigs = Enum.map(signatures, &Map.from_struct/1)
```

File: test/chainweb/pact/send_request_body_test.exs [hunk 4]
```diff
@@ -6,14 +6,16 @@ defmodule Kadena.Chainweb.Pact.SendRequestBodyTest do
...
-  alias Kadena.Types.Command
+  alias Kadena.Types.{Command, Signature}
 
   setup do
     command =
       Command.new(
         hash: "-1npoTU2Mi71pKE_oteJiJuHuXTXxoObJm8zzVRK2pk",
         sigs: [
-          "8b234b83570359e52188cceb301036a2a7b255171e856fd550cac687a946f18fbfc0e769fd8393dda44d6d04c31b531eaf35efb3b78b5e40fd857a743133030d"
+          Signature.new(
```
Comment (L-Zuluaga) [inline test/chainweb/pact/send_request_body_test.exs:16]: I was wondering why in this part you use the `.new("SIG")` and in the other tests you used the `%Signature{sign: "SIG"` notation?

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]

File: lib/types/signature.ex (hunk 1 above)
Comment (EdwinGuayacan) [reply lib/types/signature.ex]: Because they might get a `SignCommand` with the sig value as nil:
image

File: lib/pact/command/cont_command.ex (hunk 2 above)
Comment (EdwinGuayacan) [reply lib/pact/command/cont_command.ex]: Since the Signature type is returning an empty list: [ ] (when receives a nil value), and this is being concatenated with the `++` notation, it produces a list of this type [ [ ] ] which I want to avoid.

File: test/chainweb/pact/send_request_body_test.exs (hunk 4 above)
Comment (EdwinGuayacan) [reply test/chainweb/pact/send_request_body_test.exs:16]: I just tried to follow the style that each test had.

File: lib/types/signature.ex (hunk 1 above)
Comment (miguelnietoa) [reply lib/types/signature.ex]: @EdwinGuayacan the `new/1` function must return a **`Signature` struct** or an **error tuple**. 
IMO this logic behind the `nil` value regarding the `SignCommand` module should be handled in a different way. My suggestion is in the comment below ⬇️

File: lib/pact/command/cont_command.ex (hunk 2 above)
Comment (miguelnietoa) [reply lib/pact/command/cont_command.ex]: @EdwinGuayacan So I think this is just a workaround to solve that specific use case. Use a pattern matting when sig is nil and skip the creation of Signature!

File: lib/pact/command/cont_command.ex [hunk 5]
```diff
@@ -305,7 +305,9 @@ defmodule Kadena.Pact.ContCommand do
   end
 
   @spec build_signatures(sign_commands :: sign_commands(), result :: list()) :: valid_signatures()
-  defp build_signatures([], result), do: {:ok, List.flatten(result)}
+  defp build_signatures([], result), do: {:ok, result}
+
+  defp build_signatures([%SignCommand{sig: nil} | _rest], _result), do: {:ok, []}
```
Comment (miguelnietoa) [inline lib/pact/command/cont_command.ex]: @EdwinGuayacan What will happen to the remaining items on the list? Why return an empty list?

**Review by miguelnietoa — CHANGES_REQUESTED** [review]

File: lib/pact/command/cont_command.ex (hunk 5 above)
Comment (EdwinGuayacan) [reply lib/pact/command/cont_command.ex]: There are actually no remaining elements when there are no `sig`.
change to : `defp build_signatures([%SignCommand{sig: nil}], _result), do: {:ok, []}`?

**Review by miguelnietoa — APPROVED** [review]

**Review by L-Zuluaga — APPROVED** [review]
